From e38428f1f0a47ce2b28e09896e89961ec0f100a4 Thu, 1 Dec 2016 10:34:00 +0100
From: Martin Grimme <martin.grimme@gmail.com>
Date: Sun, 30 Jun 2013 10:27:52 +0200
Subject: [PATCH] properly link dynamically against the Qt5 modules... in a way that works with cmake < 2.8.8 as well


diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 807ddb4..e431876 100755
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -16,6 +16,8 @@
                       ${Qt5DBus_INCLUDE_DIRS}
                       ${Qt5Sql_INCLUDE_DIRS}
   )
+  add_definitions(${Qt5DBus_DEFINITIONS})
+  add_definitions(${Qt5Sql_DEFINITIONS})
 elseif (Qt4_FOUND)
   set(QT_USE_QTDBUS TRUE)
   set(QT_USE_QTSQL TRUE)
@@ -85,11 +87,14 @@
 
 if (Qt5_FOUND)
   add_dependencies(packagekit-qt2 mocs)
-endif ()
-
-target_link_libraries(packagekit-qt2
+  target_link_libraries(packagekit-qt2
+		${Qt5DBus_LIBRARIES} ${Qt5Sql_LIBRARIES}
+  )
+elseif (Qt4_FOUND)
+  target_link_libraries(packagekit-qt2
 		${QT_LIBRARIES}
-)
+  )
+endif ()
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/packagekit-qt2.pc.in
   ${CMAKE_CURRENT_BINARY_DIR}/packagekit-qt2.pc
