From 29e4822529f87ddea618eddb8fffb8e8f317fe1e Thu, 1 Dec 2016 10:34:27 +0100
From: Martin Grimme <martin.grimme@gmail.com>
Date: Thu, 10 Oct 2013 09:38:02 +0200
Subject: [PATCH] added method for retrieving .desktop files from cache db


diff --git a/src/transaction.cpp b/src/transaction.cpp
index 95076cd..3172858 100755
--- a/src/transaction.cpp
+++ b/src/transaction.cpp
@@ -316,36 +316,49 @@
 QString Transaction::packageIcon(const QString &packageID)
 {
     QString path;
+    QStringList desktopFiles = Transaction::packageDesktopFiles(
+                Transaction::packageName(packageID));
+
+    if (desktopFiles.size()) {
+        QFile desktopFile(desktopFiles.at(0));
+        if (desktopFile.open(QIODevice::ReadOnly | QIODevice::Text)) {
+            while (!desktopFile.atEnd()) {
+                QByteArray line = desktopFile.readLine().trimmed();
+                if (line.startsWith("Icon=")) {
+                    path = line.mid(5);
+                    break;
+                }
+            }
+            desktopFile.close();
+        } else {
+            qDebug() << "Cannot open desktop file " << desktopFile.fileName();
+        }
+    }
+
+    return path;
+}
+
+QStringList Transaction::packageDesktopFiles(const QString &packageName)
+{
+    QStringList desktopFiles;
     QSqlDatabase db = QSqlDatabase::database(PK_DESKTOP_DEFAULT_DATABASE);
     if (!db.isOpen()) {
         qDebug() << "Desktop files database is not open";
-        return path;
+        return desktopFiles;
     }
 
     QSqlQuery q(db);
     q.prepare("SELECT filename FROM cache WHERE package = :name");
-    q.bindValue(":name", Transaction::packageName(packageID));
+    q.bindValue(":name", packageName);
     if (q.exec()) {
-        if (q.next()) {
-            QFile desktopFile(q.value(0).toString());
-            if (desktopFile.open(QIODevice::ReadOnly | QIODevice::Text)) {
-                while (!desktopFile.atEnd()) {
-                    QByteArray line = desktopFile.readLine().trimmed();
-                    if (line.startsWith("Icon=")) {
-                        path = line.mid(5);
-                        break;
-                    }
-                }
-                desktopFile.close();
-            } else {
-                qDebug() << "Cannot open desktop file " << q.value(0).toString();
-            }
+        while (q.next()) {
+            desktopFiles << q.value(0).toString();
         }
     } else {
         qDebug() << "Error while running query " << q.executedQuery();
     }
 
-    return path;
+    return desktopFiles;
 }
 
 QString Transaction::lastPackage() const
@@ -763,3 +776,4 @@
 }
 
 #include "transaction.moc"
+
diff --git a/src/transaction.h b/src/transaction.h
index e9ace0a..b8efc1f 100644
--- a/src/transaction.h
+++ b/src/transaction.h
@@ -1199,6 +1199,11 @@
      */
     static QString packageIcon(const QString &packageID);
 
+    /**
+     * Returns the desktop files stored in the \p package.
+     */
+    static QStringList packageDesktopFiles(const QString &packageName);
+
 Q_SIGNALS:
     /**
      * The transaction has changed one of it's properties
