From 6b77db178211fa49c00bab58c9eec2b345f306a6 Thu, 1 Dec 2016 10:35:24 +0100
From: Sami Kananoja <sami.kananoja@jollamobile.com>
Date: Mon, 11 Nov 2013 15:22:50 +0200
Subject: [PATCH] Added method for retrieving the package name by desktop file using the desktop cache.


diff --git a/src/transaction.cpp b/src/transaction.cpp
index 3172858..740f2f9 100755
--- a/src/transaction.cpp
+++ b/src/transaction.cpp
@@ -361,6 +361,33 @@
     return desktopFiles;
 }
 
+QString Transaction::packageNameByDesktopFile(const QString &desktopFile)
+{
+    QSqlDatabase db = QSqlDatabase::database(PK_DESKTOP_DEFAULT_DATABASE);
+    if (!db.isOpen()) {
+        qDebug() << "Desktop files database is not open";
+        return QString();
+    }
+
+    QSqlQuery q(db);
+    q.prepare("SELECT package FROM cache WHERE filename = :desktopFile");
+    q.bindValue(":desktopFile", desktopFile);
+    if (q.exec()) {
+        if (q.size() == 1) {
+            q.next();
+            return q.value(0).toString();
+        } else if (q.size() > 1) {
+            qDebug() << "Multiple packages found for query " << q.executedQuery();
+        } else {
+            qDebug() << "No packages found for query " << q.executedQuery();
+        }
+    } else {
+        qDebug() << "Error while running query " << q.executedQuery();
+    }
+
+    return QString();
+}
+
 QString Transaction::lastPackage() const
 {
     Q_D(const Transaction);
diff --git a/src/transaction.h b/src/transaction.h
index b8efc1f..2fb4233 100644
--- a/src/transaction.h
+++ b/src/transaction.h
@@ -1204,6 +1204,11 @@
      */
     static QStringList packageDesktopFiles(const QString &packageName);
 
+    /**
+     * Returns the package name by the \p desktopFile
+     */
+    static QString packageNameByDesktopFile(const QString &desktopFile);
+
 Q_SIGNALS:
     /**
      * The transaction has changed one of it's properties
